src/Tribe/Widget/Widget_Interface.php

```diff
@@ -0,0 +1,226 @@
+	 * @since TBD
+	 *
+	 * @param array $arguments Set of arguments passed to the widget at hand.
+	 *
+	 * @return array<string,mixed> The parsed widget arguments map.
+	 */
+	public function parse_arguments( array $arguments );
+
+	/**
+	 * Returns the array of arguments for this widget after applying the validation callbacks.
+	 *
+	 * @since TBD
+	 *
+	 * @param array $arguments Set of arguments passed to the widget at hand.
+	 *
+	 * @return array<string,mixed> The validated widget arguments map.
+	 */
+	public function validate_arguments( array $arguments );
+
+	/**
+	 * Returns the array of callbacks for this widget's arguments.
+	 *
+	 * @since TBD
+	 *
+	 * @return array<string,mixed> A map of the widget arguments that have survived validation.
+	 */
+	public function get_validated_arguments_map();
+
+	/**
+	 * Returns a widget arguments after been parsed.
+	 *
+	 * @since TBD
+	 *
+	 * @return array<string,mixed> The widget arguments, as set by the user in the widget string.
+	 */
+	public function get_arguments();
+
+	/**
+	 * Get a widget's arguments after they have been been parsed.
+	 *
+	 * @since TBD
+	 *
+	 * @param array<string,mixed> $arguments Current set of arguments.
+	 *
+	 * @return array<string,mixed> The widget arguments, as set by the user in the widget string.
+	 */
+	public function filter_arguments( $arguments );
+
+	/**
+	 * Get a single widget argument after it has been parsed and filtered.
+	 *
+	 * @since TBD
+	 *
+	 * @param string|int   $index   Which index we intend to fetch from the arguments.
+	 * @param array|mixed  $default Default value if it doesn't exist.
+	 *
+	 * @uses  Tribe__Utils__Array::get For index fetching and Default.
+	 *
+	 * @return mixed Value for the Index passed as the first argument.
+	 */
+	public function get_argument( $index, $default = null );
+
+	/**
+	 * Filter a widget argument.
+	 *
+	 * @since TBD
+	 *
+	 * @param mixed       $argument The argument value.
+	 * @param string|int  $index    Which index we intend to fetch from the arguments.
+	 * @param array|mixed $default  Default value if it doesn't exist.
+	 *
+	 * @uses  Tribe__Utils__Array::get For index fetching and Default.
+	 *
+	 * @return mixed Value for the Index passed as the first argument.
+	 */
+	public function filter_argument( $argument, $index, $default = null );
+
+	/**
+	 * Get default arguments for a widget.
+	 *
+	 * @since TBD
+	 *
+	 * @return array<string,mixed> The map of widget default arguments.
+	 */
+	public function get_default_arguments();
+
+	/**
+	 * Get a widget's default arguments.
+	 *
+	 * @since TBD
+	 *
+	 * @param array<string,mixed> $default_arguments Current set of default arguments.
+	 *
+	 * @return array<string,mixed> The map of widget default arguments.
+	 */
+	public function filter_default_arguments( array $default_arguments = [] );
+
+	/**
+	 * Returns a widget's HTML.
+	 *
+	 * @since TBD
+	 *
+	 * @return string The rendered widget's HTML code.
+	 */
+	public function get_html();
```

Comment:
Maybe needs args passed in?

Reply:
I think the goal is to have One instance per individual widget, so it pulls from `$this->get_arguments()`